lms/djangoapps/lti_provider/users.py

```diff
@@ -92,12 +92,14 @@ def create_lti_user(lti_user_id, lti_consumer, lti_params=None):
                     edx_email = "{}@{}".format(generate_random_edx_username(), settings.LTI_USER_EMAIL_DOMAIN)
                 except User.DoesNotExist:
                     pass
-
+   
                 edx_user = User.objects.create_user(
                     username=edx_username,
                     password=edx_password,
                     email=edx_email,
                 )
+                if 'Staff' in lti_role or 'Instructor' in lti_role or 'Administrator' in lti_role:
```

Comment:
It is global permissions. As I understand you should use course permissions (see `CourseStaffRole`):
https://github.com/CredoReference/edx-platform/blob/integration-ficus/common/djangoapps/student/roles.py#L222